remove_unresponsive.py

```diff
@@ -1,38 +1,87 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python
+"""
+A simple script to remove unresponsive SpaceAPI endpoints from the directory.
+
+An entry is removed in the following cases:
+
+- Endpoint does not return HTTP 200
+
+"""
+from __future__ import print_function, division, absolute_import, unicode_literals
 
 import requests
 import json
-import collections
 
-# Get spaces list
-directory_file = './directory.json'
-directory = open(directory_file, 'r')
-spaces = json.loads(directory.read())
+# Config
+DIRECTORY_FILE = './directory.json'
+TIMEOUT_SECONDS = 10
+
+# Variables
 spaces_new = {}
+removed = []
 has_error = False
 
-# Check spaces
-for key in spaces:
-    url = spaces[key]
-    if 'spaceapi.net' in url:
-        continue
+
+def check_space(url):
+    """
+    Check a space URL. Return `true` if it's OK, or `false` if it should be
+    removed from the directory. Additionally, the reason will be returned.
+    """
+    # Fetch response
     try:
-        print '%s %s' % (key, url)
-        req = requests.get(url, verify=False, timeout=10)
-        if req.status_code == 200:
-            spaces_new[key] = url
-        else:
-            print '\t\033[0;31m\\_ Status: %s: %s\033[0m' \
-                % (req.status_code, req.reason)
-    except Exception,e:
-        print '\t\033[0;31m\\_Error: %s\033[0m' % (e)
+        response = requests.get(url, verify=False, timeout=TIMEOUT_SECONDS)
+    except requests.exceptions.ConnectTimeout:
+        return False, 'Connection timeout (%ds)' % TIMEOUT_SECONDS
+    except requests.exceptions.ReadTimeout:
+        return False, 'Read timeout (%ds)' % TIMEOUT_SECONDS
+    except requests.exceptions.ConnectionError:
+        return False, 'Connection error'
+    except Exception as e:
+        global has_error
         has_error = True
+        return False, 'Error: %s' % e
+
+    # Verify status code
+    if response.status_code != 200:
+        return False, 'Status: HTTP %s (%s)' % (response.status_code, response.reason)
+
+    # Verify JSON format
+    try:
+        data = response.json()
+    except json.decoder.JSONDecodeError:
```

Comment:
JSONDecodeError is not available in Python 2, we could just use ValueError instead.